packages/material-ui/src/Box/Box.d.ts

```diff
@@ -1,9 +1,27 @@
 import * as React from 'react';
+import { Omit } from '@material-ui/types';
+
+type ResponsiveValue<Value = string | number> =
+  | Value
+  | [Value, Value]
+  | [Value, Value, Value]
+  | [Value, Value, Value, Value]
+  | [Value, Value, Value, Value, Value]
```

Comment:
Is the union of tuples required or can we just use `Value[]`?

Reply:
We could just use `Value[]` but I preferred using tuples as it limits the number of values you can provide from 2 to 5, which is more accurate.

Reply:
People should be able to use custom breakpoints, e.g. small, medium, large, or mobile, table, desktop. I prefer @eps1lon proposal.

Reply:
So we need to be able to support more than 5 breakpoints?